### PR TITLE
Owlot/refactor use lib common

### PR DESCRIPTION
**Pull request by owlot**
This implements the new lib-common provided traits and macro's for the client-grpc code.
It also adds a mock implementation for the grpc-client.

Another pull request will follow for the server side implementations.

### CODE REVIEW

File: client-grpc/README.md [hunk 1]
```diff
@@ -1 +1,9 @@
-Files in this directory are autogenerated through the `cargo build` process.
+![Arrow Banner](https://github.com/Arrow-air/tf-github/raw/main/src/templates/doc-banner-services.png)
+
+# svc-template-rust Service
```
Comment (ServiceDog) [inline client-grpc/README.md:3]: This is the grpc client

File: client-grpc/Cargo.toml [hunk 2]
```diff
@@ -10,13 +10,35 @@ keywords.workspace     = true
...
+prost             = "0.11"
+prost-types       = "0.11"
+svc-template-rust = { path = "../server", optional = true }
+tonic             = "0.8"
+tower             = { version = "0.4", optional = true }
+
+[dependencies.lib-common]
+features = ["grpc"]
+git      = "https://github.com/Arrow-air/lib-common.git"
+tag      = "v0.1.1-develop.2"
```
Comment (ServiceDog) [inline client-grpc/Cargo.toml:28]: `develop` to keep it to latest? this tag will be updated no matter what by the devs who are working on a repo cloned from this template so it doesn't matter too much I suppose

Comment (owlot) [reply client-grpc/Cargo.toml:28]: We currently don't have a way to point to the latest develop code other than using the branch, which results in Cargo issues if the branch gets updated with a new ref.
If anything, we should update this to point to the lates release version once lib-common get's released.

File: client-grpc/README.md (hunk 1 above)
Comment (owlot) [reply client-grpc/README.md]: fixed

**Review by ServiceDog — APPROVED** [review]

**Pull request merged**